src/IsaacApiTypes.tsx

```diff
@@ -556,6 +556,11 @@ export interface UserSummaryWithEmailAddressDTO extends UserSummaryDTO {
   email?: string;
 }
 
+export interface UserSummaryWithEmailAddressAndGenderDTO extends UserSummaryDTO {
+  email?: string;
+  gender?: Gender;
```

Comment:
Just checking: can we actually have a User Summary DTO with no `gender` property?

Reply:
We have `UserSummaryDTO` which has no `email` or `gender` property (presumably for cases where we don't want to return as much personal user data).

We also have `UserSummaryWithEmailAddressDTO` which as it sounds like, extends `UserSummaryDTO` but with `email` added.

And then we have `UserSummaryWithEmailAddressAndGenderDTO` which extends `UserSummaryWithEmailAddressDTO`, adding `gender`.

Reply:
What I mean is: in the places where you use `UserSummaryWithEmailAddressAndGenderDTO`, will `gender` (and `email`) ever be `undefined`? I'm trying to understand what's the point of having this new interface (or interfaces like `UserSummaryWithEmailAddressDTO` or  `UserSummaryWithGroupMembershipDTO`) when all its properties are optional. Why not add all those to the base `UserSummaryDTO` instead? I'm not saying that's the right thing to do either, I'm just trying to understand the reasoning behind it.

Reply:
This is just the way it's been historically setup. I think there wouldn't be a functional difference to just using `UserSummaryDTO` with optional properties. However it may be to differentiate between the different locations it's used; for example on an admin endpoint we don't mind returning email addresses or gender, but in places where other students or teachers might see the response we wouldn't want to include other users emails/gender.

Reply:
Yes, that makes sense: you've created a new interface that extends another interface and adds a couple of new attributes, and this way you only use this new interface in the places were you actually want or expect a gender and an email. But my question is: if you know that you're expecting a gender and an email, why are they optional? Can they actually be `undefined`?

And another thing is that this new interface should be extending `UserSummaryWithEmailAddressDTO`.

Reply:
Arguably email can't be undefined, as you can't have an account without it. Gender could be however. Here I wasn't sure if it was better to have this extending `UserSummaryWithEmailAddressDTO` or the base `UserSummaryDTO` and opted for the latter; I'll change that now

Reply:
There might be something wider wrong with some of the typing here; making email required breaks all the tests, looks like due to mismatches in types between `UserSummaryWithEmailAddressDTO` and `UserSummaryDTO` specifically in anonymisationFunctions (and other places). I tried to do some quick fixes to it, but its not as simple as I hoped and requires checking multiple endpoints to see what the API is actually returning. Based on some commits from 5y ago it looks like the types from API were auto generated, so that would explain it! There are multiple occasions where the front end thinks it's going to receive an email address in a user summary and doesn't, or vice versa, but it hasn't flagged up as a problem because so many properties are optional. 

Created a ticket for it in the backlog, and left `email` as optional for now: https://github.com/isaaccomputerscience/isaac-cs-issues/issues/207

Reply:
The reason why `UserSummaryWithEmailAddressDTO` extends from `UserSummaryDTO` is because it's a specialisation of `UserSummaryDTO`. `UserSummaryWithEmailAddressAndGenderDTO` is a specialisation of both, so it should inherit from the less general one (`UserSummaryWithEmailAddressDTO`).

Also, thank you very much for looking into this. If the API may not return a `gender` property in some occasions, then you've done the right thing by making it optional. As long as we think about this kind of thing when creating new types/interfaces, then we're good.